### PR TITLE
DOC: Using np.histogram_bin_edges with pd.cut

### LINKED ISSUE #59165: ENH: In pd.cut(), allow bins='auto' (leveraging `np.histogram_bin_edges`)
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

While converting a quantitative variable into a qualitative one, `pd.cut()` comes in clutch. However, it requires the user to specify `bins` as either an integer or a list of bin edges. I wish it was allowed to specify `bins='auto'` similar to how `np.histogram` allows it. It internally leverages `np.histogram_bin_edges` to compute these. Thank you.

### Expectation
Instead of coding 
`pd.cut(df['x1'], bins=np.histogram_bin_edges(df['x1'], bins='auto'))`
Allow for coding
`pd.cut(df['x1'], bins='auto')`

### Additional Context

Calculation of bin edges is already done via `np.histogram_bin_edges`. Reference: https://numpy.org/doc/stable/reference/generated/numpy.histogram_bin_edges.html#numpy-histogram-bin-edges

### PR DESCRIPTION
**Pull request by chaarvii**
- [x] closes #59165 
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

As per reviews on #https://github.com/pandas-dev/pandas/pull/59241, adding documentation support for using np.histogram_bin_edges with pd.cut to allow bins calculation with strings.

### CODE REVIEW

File: pandas/core/reshape/tile.py [hunk 1]
```diff
@@ -142,12 +142,17 @@ def cut(
         fixed set of values.
     Series : One-dimensional array with axis labels (including time series).
     IntervalIndex : Immutable Index implementing an ordered, sliceable set.
+    np.histogram_bin_edges: Function to calculate only the edges of the bins
```
Comment (asishm) [inline pandas/core/reshape/tile.py:145]:
```suggestion
    numpy.histogram_bin_edges: Function to calculate only the edges of the bins
```

From the [docstring contribution docs](https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html#section-5-see-also):

> When relating to other libraries (mainly `numpy`), use the name of the module first (not an alias like `np`). If the function is in a module which is not the main one, like `scipy.sparse`, list the full module (e.g. `scipy.sparse.coo_matrix`).

**Comment by chaarvii** [comment]
@asishm Addressed your comments!

File: pandas/core/reshape/tile.py [hunk 2]
```diff
@@ -142,12 +142,17 @@ def cut(
...
     IntervalIndex : Immutable Index implementing an ordered, sliceable set.
+    numpy.histogram_bin_edges: Function to calculate only the edges of the bins
+        used by the histogram function.
 
     Notes
     -----
     Any NA values will be NA in the result. Out of bounds values will be NA in
     the resulting Series or Categorical object.
 
+    np.histogram_bin_edges can be used along with cut to calculate bins according
```
Comment (mroeschke) [inline pandas/core/reshape/tile.py:153]:
```suggestion
    ``numpy.histogram_bin_edges`` can be used along with cut to calculate bins according
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @chaarvii

**Pull request merged**